Source/GoBot/GoBot.swift

```diff
@@ -1397,13 +1397,13 @@ class GoBot: Bot {
     func markMessageAsRead(_ message: MessageIdentifier) {
         userInitiatedQueue.async {
             do {
-                let report = try self.database.report(for: message)
+                let isRead = try self.database.isReportRead(for: message)
                 try self.database.markMessageAsRead(identifier: message, isRead: true)
```

Comment:
I think this line could be moved inside the `if !isRead {` block.

Reply:
We need to call isReportRead before calling markMessageAsRead below, calling it after will yeld always true

Reply:
Right, but we only need to call `database.markMessageAsRead` if `isRead == false` right?

Wait I think I'm confused about what's going on here. The function is named `markMessageAsRead` but it checks whether the _report_ for the message is read, and then marks the message itself as read, but not the report? Why are we checking if report.isRead, and then publishing a notification that it was read even though we didn't mark it as read?

Reply:
A report is considered as read if its associated message was read, so marking a report as read is marking the message as read. What I am doing here is marking the message as read and, just for performance improvements, noitifying if a report existed for that message and the read status changed.

Just checking for is message read is not enough becuase I just want to notify for the messages that had a report associated with them.

Reply:
This makes much more sense now but I still have one more question. We don't need to call `markMessageAsRead` unless `wasAlreadyRead == false || wasAlreadyReady == nil` right? So we could move that call inside the `if`?